app/javascript/chat/message.jsx

```diff
@@ -44,31 +86,76 @@ export const Message = ({
 
   const dropdown = (
     <div className="message__actions">
-      <span className="ellipsis__menubutton">
-        <img src={ThreeDotsIcon} alt="dropdown menu icon" />
-      </span>
+      <Button
+        id={dropdownTriggerId}
+        className={`ellipsis__menubutton ${
+          dropdownOpen ? 'opacity-1' : 'opacity-0'
+        }`}
+        size="s"
+        variant="ghost"
+        onClick={() => {
+          dropdownOpen ? closeDropdownAndFocusTrigger() : setDropdownOpen(true);
+        }}
+        aria-controls={`message-options-dropdown-${id}`}
+        aria-haspopup="true"
+        aria-expanded={dropdownOpen}
+      >
+        <img src={ThreeDotsIcon} alt="Message options menu" />
+      </Button>
 
-      <div className="messagebody__dropdownmenu">
-        <Button variant="ghost" onClick={(_) => onEditMessageTrigger(id)}>
-          Edit
-        </Button>
-        <Button
-          variant="ghost-danger"
-          onClick={(_) => onDeleteMessageTrigger(id)}
-        >
-          Delete
-        </Button>
+      <div
+        id={`message-options-dropdown-${id}`}
+        className={`messagebody__dropdownmenu ${dropdownOpen ? '' : 'hidden'}`}
+      >
+        <ul class="list-none">
+          <li>
+            <Button
+              id={`edit-button-${id}`}
+              variant="ghost"
+              onClick={(_) => onEditMessageTrigger(id)}
+            >
+              Edit
+            </Button>
+          </li>
+          <li>
+            <Button
+              variant="ghost-danger"
+              onClick={(_) => onDeleteMessageTrigger(id)}
+            >
+              Delete
+            </Button>
+          </li>
+        </ul>
       </div>
     </div>
   );
   const dropdownReport = (
     <div className="message__actions">
-      <span className="ellipsis__menubutton">
-        <img src={ThreeDotsIcon} alt="message actions" />
-      </span>
+      <Button
+        id={dropdownTriggerId}
+        className={`ellipsis__menubutton ${
+          dropdownOpen ? 'opacity-1' : 'opacity-0'
```

Comment:
I'm using opacity instead of `display: none` so that the button still exists in the accessibility tree, and if a screen reader user pulls up a list of all controls on the page, this button will still appear